File: src/main/java/com/cw/flink/java/base/DateSetTransformationApp.java
package com.cw.flink.java.base;

import org.apache.flink.api.common.functions.MapFunction;
import org.apache.flink.api.java.ExecutionEnvironment;
import org.apache.flink.api.java.operators.DataSource;

import java.util.ArrayList;
import java.util.List;

public class DateSetTransformationApp {
    public static void main(String[] args) throws Exception {
        ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
        DateSetTransformationApp.mapFunction(env);
    }

    public static void mapFunction(ExecutionEnvironment env) throws Exception {
        List<Integer> list = new ArrayList<>();
        for (int i =0;i<10;i++){
            list.add(i);
        }
        DataSource<Integer> data = env.fromCollection(list);
        data.map(new MapFunction<Integer, Integer>() {
            @Override
            public Integer map(Integer value) throws Exception {
                return value+1;
            }
        }).print();
    }
}

File: src/main/scala/com/cw/flink/scala/base/DateSetTransformationApp.scala
package com.cw.flink.scala.base

import org.apache.flink.api.scala.ExecutionEnvironment
//隐式转换
import org.apache.flink.api.scala._
object DateSetTransformationApp {
  def main(args: Array[String]): Unit = {
    val env = ExecutionEnvironment.getExecutionEnvironment
    mapFuntion(env)
  }

  def mapFuntion(env: ExecutionEnvironment): Unit ={
    val data =env.fromCollection(List(1,2,3,4,5,6,7,8,9))
    //data.map((x=>x.toInt+1)).print()
    data.map(_ + 1).print()
  }
}
